### PR TITLE
K8SPXC-421: Expose mysqlx port

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPXC-421](https://jira.percona.com/browse/K8SPXC-421) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by delgod — APPROVED** [review]

**Review by qJkee — DISMISSED** [review]

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | passed |
| upgrade-haproxy | failed |
| recreate | passed |
| big-data | passed |
| storage | failed |
| scaling | passed |
| haproxy | failed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | failed |
| restore-to-encrypted-cluster | passed |
| init-deploy | failed |
| self-healing-chaos | passed |
| demand-backup | passed |
| upgrade-proxysql | failed |
| self-healing-advanced | passed |
| limits | failed |
| monitoring-2-0 | failed |
| self-healing-advanced-chaos | passed |
| pitr | passed |
| affinity | failed |
| smart-update | passed |
| operator-self-healing | passed |
| one-pod | failed |
| auto-tuning | passed |
| operator-self-healing-chaos | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |
| users | failed |
| upgrade-consistency | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/aa685567597418abd5a419369c79d9c49da9b338
image: `perconalab/percona-xtradb-cluster-operator:PR-907-aa685567`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | failed |
| scheduled-backup | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| storage | failed |
| upgrade-haproxy | failed |
| scaling-proxysql | passed |
| security-context | failed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| init-deploy | failed |
| self-healing-chaos | passed |
| demand-backup | passed |
| upgrade-proxysql | failed |
| self-healing-advanced | passed |
| limits | failed |
| monitoring-2-0 | failed |
| self-healing-advanced-chaos | passed |
| pitr | passed |
| affinity | failed |
| one-pod | failed |
| operator-self-healing | passed |
| auto-tuning | passed |
| smart-update | failed |
| operator-self-healing-chaos | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | failed |
| upgrade-consistency | failed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/b63caf226aa5323446d6e15a617fa5a9b1f17034
image: `perconalab/percona-xtradb-cluster-operator:PR-907-b63caf22`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| storage | failed |
| upgrade-haproxy | failed |
| haproxy | failed |
| scheduled-backup | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | failed |
| self-healing | passed |
| self-healing-chaos | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| init-deploy | failed |
| upgrade-proxysql | failed |
| limits | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| smart-update | passed |
| upgrade-consistency | failed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/aa685567597418abd5a419369c79d9c49da9b338
image: `perconalab/percona-xtradb-cluster-operator:PR-907-aa685567`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | failed |
| big-data | passed |
| recreate | passed |
| scaling | passed |
| scheduled-backup | failed |
| storage | passed |
| haproxy | failed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| init-deploy | failed |
| self-healing-chaos | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced-chaos | passed |
| smart-update | failed |
| monitoring-2-0 | failed |
| pitr | passed |
| operator-self-healing | passed |
| affinity | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | failed |
| users | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/b0eace06d60741e29b0d4fe57514055a84f4f232
image: `perconalab/percona-xtradb-cluster-operator:PR-907-b0eace06`

**Review by hors — CHANGES_REQUESTED** [review]
@egegunes  please fix tests and resolve the conflict
for test we need to use new haproxy image

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/b0eace06d60741e29b0d4fe57514055a84f4f232
image: `perconalab/percona-xtradb-cluster-operator:PR-907-b0eace06`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | failed |
| big-data | passed |
| scheduled-backup | passed |
| storage | passed |
| scaling | passed |
| recreate | passed |
| upgrade-haproxy | failed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | failed |
| self-healing-chaos | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| upgrade-proxysql | passed |
| limits | passed |
| monitoring-2-0 | failed |
| self-healing-advanced-chaos | passed |
| smart-update | passed |
| pitr | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing-chaos | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| upgrade-consistency | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/a6c399de5a57e9687f44eea07409f8667c7fc1da
image: `perconalab/percona-xtradb-cluster-operator:PR-907-a6c399de`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| storage | passed |
| recreate | passed |
| scheduled-backup | passed |
| haproxy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | failed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| smart-update | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | failed |
| demand-backup-encrypted-with-tls | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/256a82a569d97de73fd00df3421a1cc55a89faf4
image: `perconalab/percona-xtradb-cluster-operator:PR-907-256a82a5`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| upgrade-haproxy | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| operator-self-healing | passed |
| haproxy | passed |
| operator-self-healing-chaos | passed |
| big-data | passed |
| init-deploy | failed |
| recreate | passed |
| scheduled-backup | passed |
| scaling | passed |
| restore-to-encrypted-cluster | passed |
| scaling-proxysql | passed |
| demand-backup | passed |
| security-context | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/256a82a569d97de73fd00df3421a1cc55a89faf4
image: `perconalab/percona-xtradb-cluster-operator:PR-907-256a82a5`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | failed |
| scaling | passed |
| storage | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| big-data | passed |
| recreate | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | failed |
| affinity | passed |
| self-healing-advanced | passed |
| one-pod | passed |
| auto-tuning | passed |
| self-healing-advanced-chaos | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | passed |
| users | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/da6f61d20aeb200cba0f8728519cc645f9da3138
image: `perconalab/percona-xtradb-cluster-operator:PR-907-da6f61d2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| big-data | passed |
| scheduled-backup | passed |
| haproxy | passed |
| init-deploy | passed |
| scaling | passed |
| limits | passed |
| scaling-proxysql | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| storage | passed |
| proxysql-sidecar-res-limits | passed |
| self-healing | passed |
| users | passed |
| self-healing-chaos | passed |
| tls-issue-self | passed |
| self-healing-advanced | passed |
| tls-issue-cert-manager | passed |
| self-healing-advanced-chaos | passed |
| tls-issue-cert-manager-ref | passed |
| operator-self-healing | passed |
| validation-hook | passed |
| operator-self-healing-chaos | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/907/commits/da6f61d20aeb200cba0f8728519cc645f9da3138
image: `perconalab/percona-xtradb-cluster-operator:PR-907-da6f61d2`

File: e2e-tests/functions [hunk 1]
```diff
@@ -11,7 +11,7 @@ IMAGE=${IMAGE:-"perconalab/percona-xtradb-cluster-operator:${GIT_BRANCH}"}
 IMAGE_PXC=${IMAGE_PXC:-"perconalab/percona-xtradb-cluster-operator:main-pxc8.0"}
 IMAGE_PMM=${IMAGE_PMM:-"perconalab/pmm-client:dev-latest"}
 IMAGE_PROXY=${IMAGE_PROXY:-"perconalab/percona-xtradb-cluster-operator:main-proxysql"}
-IMAGE_HAPROXY=${IMAGE_HAPROXY:-"perconalab/percona-xtradb-cluster-operator:main-haproxy"}
+IMAGE_HAPROXY=${IMAGE_HAPROXY:-"perconalab/percona-xtradb-cluster-operator:k8spxc-421-haproxy"}
```
Comment (qJkee) [inline e2e-tests/functions:14]: do not forget to change before merge =)

File: pkg/pxc/service.go [hunk 2]
```diff
@@ -51,6 +51,13 @@ func NewServicePXC(cr *api.PerconaXtraDBCluster) *corev1.Service {
 		obj.ObjectMeta.Labels["app.kubernetes.io/component"] = appName
 		obj.ObjectMeta.Labels["app.kubernetes.io/managed-by"] = "percona-xtradb-cluster-operator"
 		obj.ObjectMeta.Labels["app.kubernetes.io/part-of"] = "percona-xtradb-cluster"
+
+		obj.Spec.Ports = append(
```
Comment (qJkee) [inline pkg/pxc/service.go:55]: i think we should guard mysqlx port definition with versions check `cr.CompareVersionWith`

**Review by qJkee — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**